Assets/_Standard Assets/Characters/Scripts/Physics/OpenCharacterController.cs

```diff
@@ -396,6 +396,12 @@ public void OnMoveLoop()
 		[SerializeField, Tooltip("Can character slide vertically when touching the ceiling? (For example, if ceiling is sloped.)")]
 		bool m_SlideAlongCeiling = true;
 
+		[SerializeField, Tooltip("Should the character keep a perfect velocity when sliding on walls?")]
+		bool m_OnWallsKeepVelocity = true;
+
+		[SerializeField, Tooltip("The maximal angle to keep a perfect velocity when sliding on walls.")]
+		float m_OnWallsMaxAngle = 10.0f;
```

Comment:
Similarly to above, I think this field should be inverted in name, and also feels a little awkwardly named to me.

How about `m_MinSlowAgainstWallsAngle`, describing the lowest angle at which we start slowing the character movement?

The current default felt pretty good to me.

Reply:
I also suggest you use the `RangeAttribute` to limit this to the (0, 90) range. The code should also properly endure the 90 case, and treat it exactly the same as if the flag above is false.